toolkit/offchain/tests/integration_tests.rs

```diff
@@ -210,15 +216,75 @@ async fn update_legacy_governance() {
+
+	let genesis_utxo =
```

Comment:
There is quite a lot going on in this test - I wonder if it would be feasible to maybe extract some pieces and get a bit closer to the common structure of setup / "action" / assertion? 

```
let something = setup_something();
let whatever = build_whatever();

let result = run_upsert_d_param(...)

assert!(result, ...)
```

Might help to make it clearer what is being asserted? WDYT?

Reply:
Of course, but this is polishing. The main question for this draft was if it is worth to build this programatically.

Reply:
Indeed, makes perfect sense. Ignore my comment :)